cocotb/share/makefiles/Makefile.inc

```
@@ -91,3 +88,33 @@ ifeq ($(OS),Msys)
 else
     to_tcl_path = $(1)
 endif
+
+# Check that the COCOTB_RESULTS_FILE was created, since we can't set an exit code from cocotb.
+define check_for_results_file
+    @test -f $(COCOTB_RESULTS_FILE) || (echo "ERROR: $(COCOTB_RESULTS_FILE) was not written by the simulation!" >&2 && exit 1)
+endef
+
+SIM_BUILD ?= sim_build
+export SIM_BUILD
+
+COCOTB_RESULTS_FILE ?= results.xml
+COCOTB_HDL_TIMEUNIT ?= 1ns
+COCOTB_HDL_TIMEPRECISION ?= 1ps
+
+# Depend on all Python from the cocotb package. This triggers a
+# recompilation of the simulation if cocotb is updated.
+CUSTOM_SIM_DEPS += $(shell find $(COCOTB_PY_DIR)/cocotb/ -name "*.py")
+
+# This triggers a recompilation of the simulation if cocotb library is updated.
+CUSTOM_SIM_DEPS += $(shell find $(LIB_DIR)/ -name "*")
+
+$(SIM_BUILD):
+	mkdir -p $@
+
+# Regression rule uses Make dependencies to determine whether to run the simulation
+.PHONY: regression
+regression: $(COCOTB_RESULTS_FILE)
+
+else
+    $(warning "Including both Makefile.inc and Makefile.sim is deprecated, as the latter now includes the former")
```

```suggestion
    $(warning "Including Makefile.inc from a user makefile is a no-op and deprecated. Remove the Makefile.inc inclusion from your makefile, and only leave the Makefile.sim include.")
```

I'd be happier if we can tighten the condition under which this deprecation message is shown to exactly the case where Makefile.sim and Makefile.inc are both included (e.g. by using a special variable COCOTB_MAKEFILE_SIM_INCLUDED or so), and not use the fact that Makefile.inc is included twice as a proxy.

---

I think to achieve this we can just move this warning to `Makefile.sim` (check for `COCOTB_MAKEFILE_INC_INCLUDED`). Should I do?   Should this be maybe `info`?